src/server/JasmineGraphInstanceService.cpp

```diff
@@ -591,8 +592,114 @@ void *instanceservicesession(void *dummyPt) {
             std::string memoryUsage = JasmineGraphInstanceService::requestPerformanceStatistics(isVMStatManager,
                                                                                                 isResourceAllocationRequired);
             write(connFd, memoryUsage.c_str(), memoryUsage.size());
+        } else if (line.compare(JasmineGraphInstanceProtocol::INITIATE_FILES) == 0) {
+            
+            instance_logger.log("Received from: " + JasmineGraphInstanceProtocol::INITIATE_FILES, "info");
+            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
+            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string trainData(data);
+
+            std::vector<std::string> trainargs = Utils::split(trainData, ' ');
+
+            string graphID;
+            string partitionID = trainargs[trainargs.size() - 1];
+
+            for (int i = 0; i < trainargs.size(); i++) {
+                if (trainargs[i] == "--graph_id") {
+                    graphID = trainargs[i + 1];
+                    break;
+                }
+            }
+
+            std::thread *workerThreads = new std::thread[2];
```

Comment:
Why specifically two threads?

Reply:
One thread allocated to the local store and the other thread is allocated to central store.

Reply:
Ok